### PR TITLE
Only allow subscribing if both users are in the same room and call.

### PR DESCRIPTION
**Pull request by fancycode**
Previously it was possible to subscribe any stream if the session id of the publisher was known.

### CODE REVIEW

**Comment by fancycode** [comment]
CC @LukasReschke @danxuliu

**Review by LukasReschke — APPROVED** [review]
Looks reasonable :) - Thanks!

**Comment by LukasReschke** [comment]
@danxuliu WDYT?

**Comment by fancycode** [comment]
@danxuliu friendly ping :-)

File: hub.go [hunk 1]
```diff
@@ -1660,6 +1691,14 @@ func (h *Hub) processMcuMessage(senderSession *ClientSession, session *ClientSes
 			return
 		}
 
+		// A user is only allowed to subscribe a stream if she is in the same room
+		// as the the user user and both have their "inCall" flag set.
```
Comment (danxuliu) [inline hub.go:1695]: Typo: `the the user user`

File: hub.go [hunk 2]
```diff
@@ -1660,6 +1691,14 @@ func (h *Hub) processMcuMessage(senderSession *ClientSession, session *ClientSes
...
+		if !h.allowSubscribeAnyStream && !h.isInSameCall(senderSession, message.Recipient.SessionId) {
+			log.Printf("Session %s is not not same call as session %s, not requesting offer", session.PublicId(), message.Recipient.SessionId)
```
Comment (danxuliu) [inline hub.go:1697]: Typo: `not not`

**Review by danxuliu — COMMENTED** [review]
Sorry a lot for the delay.

Tested (preventing the connections when the option is disabled and allowing them when the option is enabled) and works :+1:

I was wondering, would it be possible to dynamically provide the option when the publisher is created (or even in the `hello` message for the whole session) rather than making it a static option in the configuration?

Of course if that makes the code too ugly, requires much work or has any other problem the current approach is fine; it would be just a way to make stress testing in a production system easier (which, of course, might not be a good idea :-P ).

**Comment by fancycode** [comment]
> Tested (preventing the connections when the option is disabled and allowing them when the option is enabled) and works +1

Thanks! Typos fixed.

I would rather not make this configurable from the `hello` or when publishing by the clients. This is probably only useful during development and as a security feature, it should not be overridable by "modified" clients on production systems.

**Review by danxuliu — APPROVED** [review]
> I would rather not make this configurable from the hello or when publishing by the clients. This is probably only useful during development and as a security feature, it should not be overridable by "modified" clients on production systems.

OK :-)

**Pull request merged**